src/test/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSFilterTest.java

```diff
@@ -1,88 +1,66 @@
 import org.broadinstitute.hellbender.utils.bwa.BwaMemIndexCache;
 import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
 import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.SAMRecordToGATKReadAdapter;
 import org.testng.Assert;
-import org.testng.TestException;
 import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 import scala.Tuple2;
 
 import java.io.File;
-import java.io.FileReader;
-import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Comparator;
 import java.util.List;
-import java.util.stream.Collectors;
 
 public class PSFilterTest extends CommandLineProgramTest {
 
-    private final String BWA_IMAGE_PATH = "src/test/resources/" + PSFilter.class.getPackage().getName().replace(".", "/") + "/hg19mini.fasta.bwa_image";
-
-    final static List<List<String>> readFastq(final File fastqFile) {
-        try {
-            final LineIterator itr = IOUtils.lineIterator(new FileReader(fastqFile));
-            final List<List<String>> seqs = new ArrayList<>(100);
-            while (itr.hasNext()) {
-                final List<String> read = new ArrayList<>(3);
-                String readName = itr.next();
-                if (readName.endsWith("/1") || readName.endsWith("/2")) {
-                    readName = readName.substring(0, readName.length() - 2);
-                }
-                read.add(readName);
-                read.add(itr.next());
-                itr.next();
-                read.add(itr.next());
-                seqs.add(read);
-            }
-            return seqs;
-        } catch (IOException e) {
-            logger.error("Could not open test file " + fastqFile.getAbsolutePath(), e);
-        }
-        throw new TestException("Invalid fastq test file " + fastqFile);
+    private static final String BWA_IMAGE_PATH = publicTestDir + "hg19mini.fasta.img";
+
+    private static final GATKRead fastqRecordToGATKRead(final FastqRecord rec) {
+        final GATKRead read = new SAMRecordToGATKReadAdapter(new SAMRecord(null));
+        read.setName(rec.getReadName());
+        read.setBaseQualities(rec.getBaseQualities());
+        read.setBases(rec.getReadBases());
+        return read;
     }
 
-    final static List<GATKRead> getReadsFromFastq(final File fastqFile) {
-        final List<List<String>> fastqList = readFastq(fastqFile);
-        return fastqList.stream().map(list -> {
-            final String name = list.get(0);
-            final byte[] bases = list.get(1).getBytes();
-            final byte[] qual = list.get(2).getBytes();
-            final GATKRead read = ArtificialReadUtils.createArtificialRead(bases, qual, "*");
-            read.setName(name);
-            return read;
-        }).collect(Collectors.toList());
+    private static final List<GATKRead> getReadsFromFastq(final File fastqFile) {
+        final FastqReader reader = new FastqReader(fastqFile);
```

Comment:
I think you need to close the reader.... I guess you should use a try-with-resources.

Reply:
Done